src/com/vladsch/clionarduinoplugin/serial/SerialMonitorToolWindow.java

```diff
@@ -47,20 +47,22 @@ public class SerialMonitorToolWindow {
 
     public SerialMonitorToolWindow(Project project) {
         this.project = project;
-        toolWindow = ToolWindowManager.getInstance(project).registerToolWindow(TOOL_WINDOW_ID, false, ToolWindowAnchor.BOTTOM, project, true);
-        toolWindow.setIcon(PluginIcons.serial_port_tool);
-
-        //JPanel mainPanel = new JPanel(new BorderLayout());
-        //mainPanel.setBorder(BorderFactory.createEmptyBorder(0, 0, 0, 0));
-        SimpleToolWindowPanel mainPanel = new SimpleToolWindowPanel(false, true);
-
-        mySerialMonitorPanel = new SerialMonitorPanel(project);
-        mainPanel.add(mySerialMonitorPanel.getComponent());
-
-        ContentFactory contentFactory = ContentFactory.SERVICE.getInstance();
-        Content content = contentFactory.createContent(mainPanel, Bundle.message("toolwindow.serial-monitor.title"), false);
-        myToolWindowContentMap.put(SERIAL_MONITOR_ID, content);
-        toolWindow.getContentManager().addContent(content);
+        toolWindowManager.invokeLater(() -> {
+            toolWindow = ToolWindowManager.getInstance(project).registerToolWindow(TOOL_WINDOW_ID, false, ToolWindowAnchor.BOTTOM, project, true);
```

Comment:
introduce local variable

```suggestion
        ToolWindowManager toolWindowManager = ToolWindowManager.getInstance(project);
        toolWindowManager.invokeLater(() -> {
            toolWindow = toolWindowManager.registerToolWindow(TOOL_WINDOW_ID, false, ToolWindowAnchor.BOTTOM, project, true);
```

Reply:
Thank you to come up with this fix!
Uhm.. Where/how can I change an installed CLion Plugin? I can just find the CLionArduinoPlugin binary File..

Reply:
If you create a zip or jar, you may be able to install it. (I have not tried it.)
https://www.jetbrains.com/help/idea/managing-plugins.html#install_plugin_from_disk

Reply:
why is this not added 😠

Reply:
> If you create a zip or jar, you may be able to install it. (I have not tried it.) https://www.jetbrains.com/help/idea/managing-plugins.html#install_plugin_from_disk

Gives me: Fail to load plugin descriptor from file CLionArduinoPlugin-patch-1.zip